### PR TITLE
Add -d --dependency flag to get-dependencies command

### PR DESCRIPTION
**Pull request by Shellyber**
## Status
- [x] Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/46585

## Description
Added the -d --dependency flag to get-dependencies command

### CODE REVIEW

File: demisto_sdk/commands/find_dependencies/find_dependencies.py [hunk 1]
```diff
@@ -2116,6 +2118,19 @@ def find_dependencies_manager(
             dependent_packs = json.dumps(dependent_packs, indent=4)
             click.echo(click.style(dependent_packs, bold=True))
 
+        elif dependency:
+            dependent_packs, _ = get_packs_dependent_on_given_packs(input_paths, id_set_path,  # type: ignore[arg-type]
+                                                                    output_path, verbose=True)
+            dependent_items = dependent_packs[Path(input_paths[0]).name].get('packsDependentOnThisPackMandatorily')  # type: ignore[attr-defined]
+            if dependency in dependent_items:
+                print_success(f"The pack \"{dependency}\" depends on \"{Path(input_paths[0]).name}\" "
+                              f"with the following items:")
```
Comment (ShahafBenYakir) [inline demisto_sdk/commands/find_dependencies/find_dependencies.py:2127]: `input_pack_name` is depndent in `dependency` and not the other way around 
```suggestion
                print_success(f'The pack "{input_pack_name}" depends on "{dependency}" '
                              f'with the following items:')
```

Comment (ShahafBenYakir) [inline demisto_sdk/commands/find_dependencies/find_dependencies.py:2127]: `input_pack_name` is depndent in `dependency` and not the other way around 
```suggestion
                print_success(f'The pack "{input_pack_name}" depends on "{dependency}" '
                              f'with the following items:')
```

File: demisto_sdk/commands/find_dependencies/find_dependencies.py [hunk 2]
```diff
@@ -2116,6 +2118,19 @@ def find_dependencies_manager(
...
+                packs_dependencies = dependent_items.get(dependency)
+                dependencies = json.dumps(packs_dependencies, indent=4)
+                click.echo(click.style(dependencies, bold=True))
+            else:
+                print_warning("Could not find dependencies between the two packs.")
```
Comment (ShahafBenYakir) [inline demisto_sdk/commands/find_dependencies/find_dependencies.py:2132]:
```suggestion
                print_warning("Could not find dependencies between the two packs: {input_pack_name} and {dependency}")
```

Comment (ShahafBenYakir) [inline demisto_sdk/commands/find_dependencies/find_dependencies.py:2132]:
```suggestion
                print_warning("Could not find dependencies between the two packs: {input_pack_name} and {dependency}")
```
```suggestion
                print_warning("Could not find dependencies between the two packs.")
```

File: demisto_sdk/commands/find_dependencies/find_dependencies.py [hunk 3]
```diff
@@ -2116,6 +2118,19 @@ def find_dependencies_manager(
             dependent_packs = json.dumps(dependent_packs, indent=4)
             click.echo(click.style(dependent_packs, bold=True))
 
+        elif dependency:
+            dependent_packs, _ = get_packs_dependent_on_given_packs(input_paths, id_set_path,  # type: ignore[arg-type]
+                                                                    output_path, verbose=True)
+            dependent_items = dependent_packs[Path(input_paths[0]).name].get('packsDependentOnThisPackMandatorily')  # type: ignore[attr-defined]
```
[the remark this answers is not in the document]
Comment (ShahafBenYakir) [reply demisto_sdk/commands/find_dependencies/find_dependencies.py:2124]: Also let's do this as we use it later on
```suggestion
            input_pack_name = get_pack_name(input_paths[0])
            dependent_items = dependent_packs[input_pack_name].get('packsDependentOnThisPackMandatorily')  # type: ignore[attr-defined]
```

[the remark this answers is not in the document]
Comment (ShahafBenYakir) [reply demisto_sdk/commands/find_dependencies/find_dependencies.py:2124]: Also let's do this as we use it later on
```suggestion
            input_pack_name = get_pack_name(input_paths[0])
            dependent_items = dependent_packs[input_pack_name].get('packsDependentOnThisPackMandatorily')  # type: ignore[attr-defined]
```

**Review by ShahafBenYakir — COMMENTED** [review]
Look good. Missing both changelog and unit tests.

File: demisto_sdk/commands/find_dependencies/find_dependencies.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Shellyber) [reply demisto_sdk/commands/find_dependencies/find_dependencies.py:2124]: I talked about it with @dorschw as well. 
This doesn't make sense cause we want only one dependency related to another (2 packs to find the connection)
It also doesn't make sense to begin with that this is an array. 
Find dependency should be one path

**Comment by Shellyber** [comment]
Example command: ` demisto-sdk find-dependencies -i Packs/MajorBreachesInvestigationandResponse -d Packs/ServiceNow`

Example response:
```
Finished calculating the dependencies on the given packs.
The pack "MajorBreachesInvestigationandResponse" depends on "ServiceNow" with the following items:
{
    "mandatory": true,
    "dependent_items": [
        [
            [
                "incident_field",
                "incident_servicenowassignedto"
            ],
            [
                [
                    "layout",
                    "Rapid Breach Response"
                ]
            ]
        ],
        [
            [
                "incident_field",
                "incident_servicenowassignmentgroup"
            ],
            [
                [
                    "layout",
                    "Rapid Breach Response"
                ]
            ]
        ],
        [
            [
                "incident_field",
                "incident_servicenowcaller"
            ],
            [
                [
                    "layout",
                    "Rapid Breach Response"
                ]
            ]
        ],
        [
            [
                "incident_field",
                "incident_servicenowcallerid"
            ],
            [
                [
                    "layout",
                    "Rapid Breach Response"
                ]
            ]
        ],
        [
            [
                "incident_field",
                "incident_servicenowcategory"
            ],
            [
                [
                    "layout",
                    "Rapid Breach Response"
                ]
            ]
        ],
        [
            [
                "incident_field",
                "incident_servicenowclosedby"
            ],
            [
                [
                    "layout",
                    "Rapid Breach Response"
                ]
            ]
        ],
        [
            [
                "incident_field",
                "incident_servicenowcloseddate"
            ],
            [
                [
                    "layout",
                    "Rapid Breach Response"
                ]
            ]
        ],
        [
            [
                "incident_field",
                "incident_servicenowdescription"
            ],
            [
                [
                    "layout",
                    "Rapid Breach Response"
                ]
            ]
        ],
        [
            [
                "incident_field",
                "incident_servicenowduedate"
            ],
            [
                [
                    "layout",
                    "Rapid Breach Response"
                ]
            ]
        ],
        [
            [
                "incident_field",
                "incident_servicenowescalation"
            ],
            [
                [
                    "layout",
                    "Rapid Breach Response"
                ]
            ]
        ],
        [
            [
                "incident_field",
                "incident_servicenowimpact"
            ],
            [
                [
                    "layout",
                    "Rapid Breach Response"
                ]
            ]
        ],
        [
            [
                "incident_field",
                "incident_servicenownotify"
            ],
            [
                [
                    "layout",
                    "Rapid Breach Response"
                ]
            ]
        ],
        [
            [
                "incident_field",
                "incident_servicenowopeneddate"
            ],
            [
                [
                    "layout",
                    "Rapid Breach Response"
                ]
            ]
        ],
        [
            [
                "incident_field",
                "incident_servicenowpriority"
            ],
            [
                [
                    "layout",
                    "Rapid Breach Response"
                ]
            ]
        ],
        [
            [
                "incident_field",
                "incident_servicenowresolutioncode"
            ],
            [
                [
                    "layout",
                    "Rapid Breach Response"
                ]
            ]
        ],
        [
            [
                "incident_field",
                "incident_servicenowresolutionnotes"
            ],
            [
                [
                    "layout",
                    "Rapid Breach Response"
                ]
            ]
        ],
        [
            [
                "incident_field",
                "incident_servicenowresolvedtime"
            ],
            [
                [
                    "layout",
                    "Rapid Breach Response"
                ]
            ]
        ],
        [
            [
                "incident_field",
                "incident_servicenowseverity"
            ],
            [
                [
                    "layout",
                    "Rapid Breach Response"
                ]
            ]
        ],
        [
            [
                "incident_field",
                "incident_servicenowstate"
            ],
            [
                [
                    "layout",
                    "Rapid Breach Response"
                ]
            ]
        ],
        [
            [
                "incident_field",
                "incident_servicenowticketnumber"
            ],
            [
                [
                    "layout",
                    "Rapid Breach Response"
                ]
            ]
        ],
        [
            [
                "incident_field",
                "incident_servicenowurgency"
            ],
            [
                [
                    "layout",
                    "Rapid Breach Response"
                ]
            ]
        ]
    ]
}

```

File: CHANGELOG.md [hunk 4]
```diff
@@ -22,6 +22,8 @@ the same as the last release note version.
 * Added a new command **generate-unit-tests**.
 * Fixed the destination path of the unified parsing/modeling rules in **create-content-artifacts** command.
 * Fixed an issue in the **validate** command, where we validated wrongfully the existence of readme file for the *ApiModules* pack.
+* Added the flag '-d', '--dependency' to **find-dependencies** to get the content items that cause the dependencies between two packs.
```
Comment (ShahafBenYakir) [inline CHANGELOG.md:25]:
```suggestion
* Added the flag '-d', '--dependency' to **find-dependencies** command to get the content items that cause the dependencies between two packs.
```

**Review by ShahafBenYakir — COMMENTED** [review]
Looks good.
Let's add tests for this.

**Comment by Shellyber** [comment]
@ShahafBenYakir Done

File: demisto_sdk/commands/find_dependencies/tests/find_dependencies_test.py [hunk 5]
```diff
@@ -3178,6 +3178,34 @@ def test_get_dependent_on_given_pack(self, mocker):
         assert dependent_packs_dict['pack3']['packsDependentOnThisPackMandatorily']['pack2']['dependent_items'] == [
             (('type_item_3', 'item3'), ('type_item_b', 'item_b'))]
 
+    def test_find_dependencies_between_two_packs(self, mocker):
```
Comment (ShahafBenYakir) [inline demisto_sdk/commands/find_dependencies/tests/find_dependencies_test.py:3181]: Add given when then.

**Review by ShahafBenYakir — APPROVED** [review]
Let's just add unit tests docstrings

**Pull request merged**